warcprox/mitmproxy.py

```diff
@@ -637,13 +637,17 @@ def status(self):
 
     def process_request(self, request, client_address):
         self.active_requests[request] = doublethink.utcnow()
-        future = self.pool.submit(
-                self.process_request_thread, request, client_address)
-        future.add_done_callback(
-                lambda f: self.active_requests.pop(request, None))
-        if future.done():
-            # avoid theoretical timing issue, in case process_request_thread
-            # managed to finish before future.add_done_callback() ran
+        try:
+            future = self.pool.submit(
+                    self.process_request_thread, request, client_address)
+            future.add_done_callback(
+                    lambda f: self.active_requests.pop(request, None))
+            if future.done():
+                # avoid theoretical timing issue, in case process_request_thread
+                # managed to finish before future.add_done_callback() ran
+                self.active_requests.pop(request, None)
+        except RuntimeError as exc:
+            self.logger.error("Error processing request %s", str(exc))
```

Comment:
you shouldn't call `str(exc)` that happens automatically because of `%s` if the error is logged and doesn't make that unneeded computation if it is filtered.

Additionally you could instead do, `self.logger.error("Error processing request", exc_info=exc)` than logging will add information about the exception by itself.

Before py3.5 `sys.exc_info()` will be used as `bool(exc)` is true
and after `exc` will be used directly.

https://docs.python.org/3/library/logging.html#logging.Logger.debug